File: C_Programs/FactorialUsingDoWhileLoop.cpp
// FactorialUsingDoWhileLoop.cpp : This file contains the 'main' function. Program execution begins and ends there.
//

#include <iostream>

int main()
{
	int num, fact=1;
	printf("Enter a number : ");
	scanf_s("%d", &num);
	do
	{
		fact = num * fact;
		num--;
	} while (num > 0);
	printf("\nmFactorial of the number is : %d", fact);
	return 0;
}

File: Basic_Java_Programs/FactorialUsingDoWhileLoop.java
//factorial of a number using do while loop
import java.util.*;

public class FactorialUsingDoWhileLoop {
    public static void main(String[] args){
        int num, fact=1;
        int temp;
        Scanner sc=new Scanner(System.in);

        System.out.println("Enter a number : ");
        num=sc.nextInt();
        temp=num;

        do {
            fact = num * fact;
            num--;
        } while (num > 0);
        System.out.println("Factorial of "+temp+" is : "+fact);
    }
}
